src/py_string_cache.rs

```diff
@@ -183,3 +184,19 @@ impl PyStringCache {
         self.entries.fill(None);
     }
 }
+
+pub fn pystring_fast_new<'py>(py: Python<'py>, s: &str, ascii_only: bool) -> Bound<'py, PyString> {
+    if ascii_only {
+        unsafe { pystring_ascii_new(py, s) }
+    } else {
+        PyString::new_bound(py, s)
+    }
+}
+
+unsafe fn pystring_ascii_new<'py>(py: Python<'py>, s: &str) -> Bound<'py, PyString> {
+    let ptr = ffi::PyUnicode_New(s.len() as isize, 127);
+    let data_ptr = ptr.cast::<ffi::PyASCIIObject>().offset(1) as *mut u8;
```

Comment:
Can we perhaps consider using `PyUnicode_DATA` here? Or at least asserting that `PyUnicode_KIND == PyUnicode_1BYTE_KIND`? Ideally both.

I feel like here we're making a huge assumption that CPython doesn't change the memory representation of `PyUnicode_New(len, 127)` and if that assumption breaks we're 100% in memory corruption territory here.

Reply:
Done.

@davidhewitt can you check one more time that you're ~~happy~~ less unhappy with it now? In particular, I'd like to confirm I'm using `PyUnicode_DATA` correctly.